sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3227,6 +3227,60 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       Option(ctx.partitionSpec).map(visitNonOptionalPartitionSpec))
   }
 
+  /**
+   * Create or replace a view. This creates a [[CreateViewStatement]]
+   *
+   * For example:
+   * {{{
+   *   CREATE [OR REPLACE] [[GLOBAL] TEMPORARY] VIEW [IF NOT EXISTS] multi_part_name
+   *   [(column_name [COMMENT column_comment], ...) ]
+   *   create_view_clauses
+   *
+   *   AS SELECT ...;
+   *
+   *   create_view_clauses (order insensitive):
+   *     [COMMENT view_comment]
+   *     [TBLPROPERTIES (property_name = property_value, ...)]
+   * }}}
+   */
+  override def visitCreateView(ctx: CreateViewContext): LogicalPlan = withOrigin(ctx) {
+    if (!ctx.identifierList.isEmpty) {
+      operationNotAllowed("CREATE VIEW ... PARTITIONED ON", ctx)
+    }
+
+    checkDuplicateClauses(ctx.COMMENT, "COMMENT", ctx)
+    checkDuplicateClauses(ctx.PARTITIONED, "PARTITIONED ON", ctx)
+    checkDuplicateClauses(ctx.TBLPROPERTIES, "TBLPROPERTIES", ctx)
+
+    val userSpecifiedColumns = Option(ctx.identifierCommentList).toSeq.flatMap { icl =>
+      icl.identifierComment.asScala.map { ic =>
+        ic.identifier.getText -> Option(ic.STRING).map(string)
+      }
+    }
+
+    val persistedView: Int = 1
+    val globalTempView: Int = 2
+    val localTempView: Int = 3
+    val viewType = if (ctx.TEMPORARY == null) {
+      persistedView
+    } else if (ctx.GLOBAL != null) {
+      globalTempView
+    } else {
+      localTempView
+    }
```

Comment:
I can't access ```PersistedView/GlobalTempView/LocalTempView``` in this class. I use 1/2/3 and map them back to ```PersistedView/GlobalTempView/LocalTempView``` in ```ResolveSessionCatalog```

Reply:
Can we move PersistedView etc.?

Reply:
Yes. We can move these to catalyst. What is a good place to put these? ```org.apache.spark.sql.catalyst.analysis.view.scala```? ```org.apache.spark.sql.catalyst.catalog.interface.scala```?

Reply:
I think view.scala is better. cc @cloud-fan

Reply:
+1 for view.scala